### PR TITLE
Ticket #3334 - Improved error reporting/handling when reading raw data files; use processed files only if the user does not select 'Use raw files if possible' tick box when loading the experiment.

### PR DESCRIPTION
**Pull request by rpetry**
@nsklyar - could you review please

### CODE REVIEW

File: atlas-loader/src/main/java/uk/ac/ebi/gxa/loader/steps/ArrayDataStep.java [hunk 1]
```diff
@@ -234,27 +231,21 @@ public boolean readArrayData(@Nonnull AtlasComputeService computeService, MAGETA
                                 localZipFile = File.createTempFile("atlas-loader", ".zip");
                                 zipFiles.put(zipName, localZipFile);
                                 copyFile(new URL(zipName), localZipFile);
+                                extractZip(localZipFile, adData.dataDir);
```
Comment (nsklyar) [inline atlas-loader/src/main/java/uk/ac/ebi/gxa/loader/steps/ArrayDataStep.java]: currently extractZip(localZipFile, adData.dataDir) called only  "if (localZipFile == null) " in the original version it was called outside of this condition.

File: atlas-loader/src/main/java/uk/ac/ebi/gxa/loader/steps/ArrayDataStep.java [hunk 2]
```diff
@@ -234,27 +231,21 @@ public boolean readArrayData(@Nonnull AtlasComputeService computeService, MAGETA
...
-                            return false;
-                        }
                     }
                 }
+
                 if (!tempFile.exists()) {
-                    log.warn("File '" + dataFileName + "' is not found");
-                    return false;
+                    log.error("File '" + dataFileName + "' is not found");
+                    throw new AtlasLoaderException("Error occurred while processing raw data files" + USE_PROCCESSED_FILES);
```
Comment (nsklyar) [inline atlas-loader/src/main/java/uk/ac/ebi/gxa/loader/steps/ArrayDataStep.java]: do we need to log.error just before to throw exception here and other places in the method?

File: atlas-loader/src/main/java/uk/ac/ebi/gxa/loader/service/AtlasMAGETABLoader.java [hunk 3]
```diff
@@ -129,18 +129,14 @@ public void process(LoadExperimentCommand cmd, AtlasLoaderServiceListener listen
...
                 if (useRawData != null && useRawData.size() == 1 && "true".equals(useRawData.iterator().next())) {
                     logProgress(listener, 5, ArrayDataStep.displayName());
-                    arrayDataRead = new ArrayDataStep().readArrayData(atlasComputeService, investigation, listener, cache);
-                }
-
-                logProgress(listener, 6, DerivedArrayDataMatrixStep.displayName());
-                if (arrayDataRead) {
+                    new ArrayDataStep().readArrayData(atlasComputeService, investigation, listener, cache);
                     log.info("Raw data are used; processed data will not be processed");
                 } else {
```
Comment (nsklyar) [inline atlas-loader/src/main/java/uk/ac/ebi/gxa/loader/service/AtlasMAGETABLoader.java:138]: maybe to add here comment "use processed data" to be consistent with the rest of method.

File: atlas-loader/src/main/java/uk/ac/ebi/gxa/loader/steps/ArrayDataStep.java (hunk 1 above)
Comment (rpetry) [inline atlas-loader/src/main/java/uk/ac/ebi/gxa/loader/steps/ArrayDataStep.java]: Thanks - should be fixed now.

File: atlas-loader/src/main/java/uk/ac/ebi/gxa/loader/steps/ArrayDataStep.java (hunk 2 above)
Comment (rpetry) [inline atlas-loader/src/main/java/uk/ac/ebi/gxa/loader/steps/ArrayDataStep.java]: Moved the error msg to the AtlasLoaderException

**Pull request merged**